connector-runtime/connector-runtime-core/src/test/java/io/camunda/connector/runtime/core/outbound/ConnectorJobHandlerTest.java

```diff
@@ -427,14 +472,130 @@ void shouldTruncateBpmnErrorMessage() {
                       "testProperty",
                       ERROR_EXPRESSION_KEYWORD,
                       errorExpression))
-              .execute(jobHandler, false, true);
+              .executeAndCaptureResult(jobHandler, false, true);
 
       // then
       assertThat(result.getErrorMessage().length())
           .isLessThanOrEqualTo(ConnectorJobHandler.MAX_ERROR_MESSAGE_LENGTH);
     }
   }
 
+  @Nested
+  class RetryBackoffTests {
+
+    private FailJobCommandStep1 firstStepMock;
+    private FailJobCommandStep2 secondStepMock;
+    private JobClient jobClient;
```

Comment:
Could we replacde/enhance the tests with a zeebe integration test that shows the retries (by incrementing a or counter or something)?

Its great we have a test setup based on mocking but in this case a e2e integration test would be great.